AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -1143,57 +1136,32 @@ - (void)nodeDidRelayout:(ASCellNode *)node sizeChanged:(BOOL)sizeChanged
     return;
   }
   
-  BOOL queued = (_queuedNodeSizeInvalidationContext != nil);
-  if (!queued) {
-    _queuedNodeSizeInvalidationContext = [[_ASCollectionViewNodeSizeInvalidationContext alloc] init];
-    
-    __weak __typeof__(self) weakSelf = self;
-    dispatch_async(dispatch_get_main_queue(), ^{
-      __typeof__(self) strongSelf = weakSelf;
-      if (strongSelf) {
-        [strongSelf requeryNodeSizes];
-      }
-    });
+  NSIndexPath *indexPath = [self indexPathForNode:node];
+  if (indexPath == nil) {
+    return;
   }
+
+  [_layoutFacilitator collectionViewWillEditCellsAtIndexPaths:@[ indexPath ] batched:NO];
   
-  [_queuedNodeSizeInvalidationContext.invalidatedNodes addObject:node];
+  ASCollectionViewInvalidationAction previousAction = _nextLayoutInvalidationAction;
+  ASCollectionViewInvalidationAction action = previousAction;
```

Comment:
The way this is done is confusing.  I think we should probably do action = _nextLayout, and then previousAction = action, although even that is a bit strange.  I see why it is needed though.